src/snowflake/cli/api/secure_path.py

```diff
@@ -0,0 +1,246 @@
+
+    def read_text(self, file_size_limit_kb: int, encoding=None, errors=None) -> str:
+        """
+        Return the decoded contents of the file as a string.
+        Raises an error of the file exceeds the specified size limit.
+        For details, check pathlib.Path.read_text()
+        """
+        self._assert_exists_and_is_file()
+        self._assert_file_size_limit(file_size_limit_kb)
+        log.info("Reading file %s", self._path)
+        return self._path.read_text(encoding=encoding, errors=errors)
+
+    @contextmanager
+    def open(  # noqa: A003
+        self,
+        mode="r",
+        read_file_limit_kb: Optional[int] = None,
+        buffering=-1,
+        encoding=None,
+        errors=None,
+        newline=None,
+    ):
+        """
+        Open the file pointed by this path and return a file object, as
+        the built-in open() function does.
+        If the file is opened for reading, [read_file_limit_kb] parameter must be provided.
+        Raises error if the read file exceeds the specified size limit.
+        """
+        opened_for_reading = "r" in mode
+        if opened_for_reading:
+            assert (
+                read_file_limit_kb is not None
+            ), "For reading mode ('r') read_file_limit_kb must be provided"
+            self._assert_exists_and_is_file()
+            self._assert_file_size_limit(read_file_limit_kb)
+
+        if self.exists():
+            self._assert_is_file()
+        else:
+            self.touch()  # makes sure permissions of freshly-created file are strict
+
+        log.info("Opening file %s in mode '%s'", self._path, mode)
+        with self._path.open(
+            mode=mode,
+            buffering=buffering,
+            encoding=encoding,
+            errors=errors,
+            newline=newline,
+        ) as fd:
+            yield fd
+        log.info("Closing file %s", self._path)
+
+    def copy(self, destination: Union[Path, str]) -> "SecurePath":
+        """
+        Copy the file/directory into the destination.
+        If source is a directory, its whole content is copied recursively.
+        Permissions of the copy are limited only to the owner.
+
+        If destination is an existing directory, the copy will be created inside it.
+        Otherwise, the copied file/base directory will be renamed to match destination.
+        If the destination file/directory already exists, FileExistsError will be raised.
+        """
+        self._assert_exists()
+
+        destination = Path(destination)
+        if destination.exists():
+            if destination.is_dir():
+                destination = destination / self._path.name
+            if destination.exists():
+                raise FileExistsError(destination)
+
+        def _recursive_copy(src: Path, dest: Path):
+            if src.is_file():
+                shutil.copyfile(src, dest)
+            if src.is_dir():
+                dest.mkdir(mode=0o700)
```

Comment:
not 600?

Reply:
"exec" for directory means "can access elements inside directory" (ls, cd), so user should have it enabled
https://unix.stackexchange.com/questions/21251/execute-vs-read-bit-how-do-directory-permissions-in-linux-work